pkg/flink/handler.go

```diff
@@ -135,6 +136,25 @@ func (flinkResourceHandler) BuildIdentityResource(ctx context.Context, taskCtx p
 	}, nil
 }
 
+func (h flinkResourceHandler) OnAbort(ctx context.Context, tCtx pluginsCore.TaskExecutionContext, resource client.Object) (behavior k8s.AbortBehavior, err error) {
+	flinkCluster := resource.(*flinkOp.FlinkCluster)
+
+	if flinkCluster.Status.Components.Job != nil && flinkCluster.Status.Components.Job.State == flinkOp.JobStatePending {
```

Comment:
since this is a "job update", the operator checks that there is a valid `savepointsDir` defined in the jobSpec, and fails if not. Should we check for that here--and if it's not set simply delete the resource?